File: igVortex/Python/src/meshes/igcamberMESH.py
import math
import numpy as np

def igcamberMESH(c_, d_, camber):
    """
    Generate a mesh surrounding a cambered airfoil.

    Input:
    * c_: chord length
    * d_: stroke length:
    * camber

    Output:
    * ZETA: mesh points in airfoil fixed system

    Local Variables:
    * dX: x-increment
    * dY: y-increment
    * maxX: x-max
    * maxY: y-max
    """

    dX = 0.2 * c_
    dY = 0.2 * c_
    maxX = 1.0 * d_
    maxY = 1.0 * d_

    x1 = np.linspace(-0.5, 0.5, dX)
    x2 = np.linspace(0.7, maxX, dX)
    x3 = -np.fliplr(x2)
    x = np.append(x3, [x1, x2])
    nx = x.shape[0]
    atmp_ = 0.5
    y1 = camber * (atmp_ ** 2 )
    y2 = 0.0 * x2
    y = np.append(y2, [y1, y2])
    nyh = math.floor(nx / 2)

    for i in range(nyh):
        xi[i+nyh,:] = x
        eta[i+nyh,:] = y + (i - 0.5) * dY
        xi[i, :] = x
        eta[i, :] = y - (nyh - i + 0.5) * dY

    ZETA = complex(xi, eta)
    return ZETA / d_

File: 2D Simulation/MATLAB/meshes/igcamberMESH.m
function [ZETA ] = igcamberMESH(c_, d_, camber)
%--------------------------------------------------------------------------
%Generate a mesh surrounding a cambered airfoil 

%INPUT
% c_    chord length
% d_    stroke length
%OUTPUT:
    %ZETA       mesh points in airfoil fixed system; size(ZETA)=(m,n)
%LOCAL VALUABLES:
    %dX     x-increment
    %dY     y-increment
    %maxX   x-max
    %maxY   y-max
%--------------------------------------------------------------------------

    dX  =0.20*c_;
    dY  =0.20*c_;
    maxX=1.0*d_;
    maxY=1.0*d_;
    x1=-0.5:dX:0.5;
    x2= 0.7:dX:maxX;
    x3=-fliplr(x2);
    x=[x3 x1 x2];
    nx=length(x);
    atmp_=0.5;
    y1=camber*(atmp_^2-x1.^2);
    y2=0.0*x2;
    y=[y2 y1 y2];
    nyh=floor(nx/2);
        
    for i=1:nyh
        xi(i+nyh,:)=x;
        eta(i+nyh,:)=y+(i-0.5)*dY;
        xi(i    ,:)=x;
        eta(i    ,:)=y-(nyh-i+0.5)*dY;
    end
    ZETA=complex(xi,eta);
    %nondimentionalize the mesh
    ZETA=ZETA/d_;  
end

